File: src/main/java/entities/gestionStore/produit.java
package entities.gestionStore;

public class produit {

    private int idProduit;
    private String name;
    private float prix;
    private int stock;
    private String description;
    private String categorie;
    private String photo;
    private int seuil;
    private float promo;

    //getters and setters
    public int getIdProduit() {
        return idProduit;
    }

    public void setIdProduit(int idProduit) {
        this.idProduit = idProduit;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public float getPrix() {
        return prix;
    }

    public void setPrix(float prix) {
        this.prix = prix;
    }

    public int getStock() {
        return stock;
    }

    public void setStock(int stock) {
        this.stock = stock;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getCategorie() {
        return categorie;
    }

    public void setCategorie(String categorie) {
        this.categorie = categorie;
    }

    public String getPhoto() {
        return photo;
    }

    public void setPhoto(String photo) {
        this.photo = photo;
    }

    public int getSeuil() {
        return seuil;
    }

    public void setSeuil(int seuil) {
        this.seuil = seuil;
    }

    public float getPromo() {
        return promo;
    }

    public void setPromo(float promo) {
        this.promo = promo;
    }
    //const param et non


    public produit(int idProduit, String name, float prix, int stock, String description, String categorie, String photo, int seuil, float promo) {
        this.idProduit = idProduit;
        this.name = name;
        this.prix = prix;
        this.stock = stock;
        this.description = description;
        this.categorie = categorie;
        this.photo = photo;
        this.seuil = seuil;
        this.promo = promo;
    }

    public produit() {
    }

    //tous les att sauf idProduit car auto increment

    public produit(String name, float prix, int stock, String description, String categorie, String photo, int seuil, float promo) {
        this.name = name;
        this.prix = prix;
        this.stock = stock;
        this.description = description;
        this.categorie = categorie;
        this.photo = photo;
        this.seuil = seuil;
        this.promo = promo;
    }


    //toString


    @Override
    public String toString() {
        return "produit{" +
                "idProduit=" + idProduit +
                ", name='" + name + '\'' +
                ", prix=" + prix +
                ", stock=" + stock +
                ", description='" + description + '\'' +
                ", categorie='" + categorie + '\'' +
                ", photo='" + photo + '\'' +
                ", seuil=" + seuil +
                ", promo=" + promo +
                '}';
    }
}

File: webapp/src/GymPlus/src/Entity/Produit.php
<?php

namespace App\Entity;

use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use App\Repository\ProduitRepository;

#[ORM\Entity(repositoryClass: ProduitRepository::class)]
class Produit
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $idproduit;

    #[ORM\Column]
    private ?string $name;

    #[ORM\Column]
    private ?float $prix;

    #[ORM\Column]
    private ?int $stock;

    #[ORM\Column]
    private ?string $description;

    #[ORM\Column]
    private ?string $categorie;


    #[ORM\Column]
    private ?string $photo;

    #[ORM\Column]
    private ?int $seuil;

    #[ORM\Column]
    private ?float $promo;

    public function getIdproduit(): ?int
    {
        return $this->idproduit;
    }

    public function getName(): ?string
    {
        return $this->name;
    }

    public function setName(?string $name): static
    {
        $this->name = $name;

        return $this;
    }

    public function getPrix(): ?float
    {
        return $this->prix;
    }

    public function setPrix(?float $prix): static
    {
        $this->prix = $prix;

        return $this;
    }

    public function getStock(): ?int
    {
        return $this->stock;
    }

    public function setStock(?int $stock): static
    {
        $this->stock = $stock;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(?string $description): static
    {
        $this->description = $description;

        return $this;
    }

    public function getCategorie(): ?string
    {
        return $this->categorie;
    }

    public function setCategorie(?string $categorie): static
    {
        $this->categorie = $categorie;

        return $this;
    }

    public function getPhoto(): ?string
    {
        return $this->photo;
    }

    public function setPhoto(?string $photo): static
    {
        $this->photo = $photo;

        return $this;
    }

    public function getSeuil(): ?int
    {
        return $this->seuil;
    }

    public function setSeuil(?int $seuil): static
    {
        $this->seuil = $seuil;

        return $this;
    }

    public function getPromo(): ?float
    {
        return $this->promo;
    }

    public function setPromo(?float $promo): static
    {
        $this->promo = $promo;

        return $this;
    }


}
